builtin/credential/ldap/backend.go

```diff
@@ -206,13 +154,11 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 		return username, policies, ldapResponse, allGroups, nil
 	}
 
-	entityAliasAttribute, err := ldapClient.GetUserAliasAttributeValue(cfg.ConfigEntry, c, username)
-	if err != nil {
-		return "", nil, logical.ErrorResponse(err.Error()), nil, nil
-	}
-	if entityAliasAttribute == "" {
+	userAttrValues := c.UserAttributes[cfg.UserAttr]
```

Comment:
UUIC `c.UserAttributes[cfg.UserAttr]` is set from cap/ldap `getUserBindDN()`? Would be worth double-checking that the cap/ldap implementation would result in the same entity alias as the prior code in `GetUserAliasAttributeValue()`. We've have entity alias changes result in privilege escalation in the past so worth a look!

Reply:
Yeah it seems like the cap/ldap implementation doesn't apply the `userfilter` but I'm not sure if that would matter since the `userfilter` would've already been applied from the search to get the user bind dn.

But based on the existing client, https://github.com/hashicorp/vault/blob/be83f12fff085b897638900802cdde190279c08b/sdk/helper/ldaputil/client.go#L149 it does like it uses `cfg.UserAttr` as well.